pkg/app/run.go

```diff
@@ -123,6 +131,48 @@ func Run(opts RunOpts) error {
 		}
 	}
 
+	if data.File.Profile != "" {
+		if name, p := profile.Get(data.File.Profile, globals.File.Profiles); name != "" {
+			token = p.Token
+		}
+	}
+
+	if globals.Flag.Profile != "" && command.Name() != "configure" {
+		if exist := profile.Exist(globals.Flag.Profile, globals.File.Profiles); exist {
+			// Persist the permanent switch of profiles.
+			var ok bool
+			if globals.File.Profiles, ok = profile.Set(globals.Flag.Profile, globals.File.Profiles); ok {
+				if err := globals.File.Write(globals.Path); err != nil {
+					return err
+				}
+			}
+		} else {
+			msg := profile.DoesNotExist
+
+			name, p := profile.Default(globals.File.Profiles)
+			if name == "" {
+				msg = fmt.Sprintf("%s (no account profiles configured)", msg)
+				return fsterr.RemediationError{
+					Inner:       fmt.Errorf(msg),
+					Remediation: fsterr.ProfileRemediation,
+				}
+			}
+
+			msg = fmt.Sprintf("%s The default profile '%s' (%s) will be used.", msg, name, p.Email)
+			text.Warning(opts.Stdout, msg)
+
+			label := "\nWould you like to continue? [y/N] "
+			cont, err := text.AskYesNo(opts.Stdout, label, opts.Stdin)
+			if err != nil {
+				return err
+			}
+			if !cont {
+				return nil
+			}
+			token = p.Token
+		}
+	}
+
```

Comment:
**TODO**: Not essential for the PR review. 

I'm thinking of refactoring this chunk of code into the new `profile` package.

Reply:
This refactor is now pushed: https://github.com/fastly/cli/pull/546/files/#diff-7e180686f4acd352dbcfd7ac75355f83218e23e9dd38699bb0ac8f9029903175R95